visualization/dashboard/src/MLIDashboard/Controls/FilterEditor.tsx

```diff
@@ -244,49 +386,88 @@ export class FilterEditor extends React.PureComponent<IFilterEditorProps, IFilte
-        const selectedVals = [...(this.state.arg as number[])];
+        const openedFilter = this.state.openedFilter;
+        const selectedVals = [...(openedFilter.arg as number[])];
+
         const index = selectedVals.indexOf(item.key as number);
         if (item.selected && index === -1) {
             selectedVals.push(item.key as number);
         } else {
             selectedVals.splice(index, 1);
         }
-        this.setState({arg: selectedVals});
+        this.setState(
+            {
+                openedFilter:{
+                    arg:selectedVals,
+                    method:openedFilter.method,
+                    column:openedFilter.column 
+                }
+            }
+        )
     }
 
     private readonly setComparison = (event: React.FormEvent<IComboBox>, item: IComboBoxOption): void => {
-        this.setState({method: item.key as FilterMethods});
+       const openedFilter = this.state.openedFilter;
+        this.setState(
+            {
+                openedFilter:{
+                    arg:openedFilter.arg,
+                    method:item.key as FilterMethods,
+                    column:openedFilter.column 
+                }
+            }
+        )
     }
 
     private readonly setNumericValue = (delta: number, column: IJointMeta, stringVal: string): string | void => {
+        const openedFilter = this.state.openedFilter;
         if (delta === 0) {
             const number = +stringVal;
             if ((!Number.isInteger(number) && column.featureRange.rangeType === RangeTypes.integer)
                 || number > column.featureRange.max || number < column.featureRange.min) {
-                return this.state.arg.toString();
+                return this.state.openedFilter.arg.toString();
             }
-            this.setState({arg: number});
+
+            this.setState(
+                {
+                    openedFilter:{
+                        arg:number,
+                        method:openedFilter.method,
+                        column:openedFilter.column 
+                    }
+                }
+            )
         } else {
-            const prevVal = this.state.arg as number;
+            const prevVal = openedFilter.arg as number;
             const newVal = prevVal + delta;
             if (newVal > column.featureRange.max || newVal < column.featureRange.min) {
                 return prevVal.toString();
             }
-            this.setState({arg: newVal});
+            this.setState(
+                {
+                    openedFilter:{
+                        arg:newVal,
+                        method:openedFilter.method,
+                        column:openedFilter.column 
+                    }
+                }
+            )
         }
     }
 
     private setDefaultStateForKey(key: string): void {
-        const filter: IFilter = {column: key} as IFilter;
+        const openedFilter = this.state.openedFilter;
```

Comment:
The currently opened filter is not needed here, the user has clicked on a different item in the left panel, so they are changing away from the existing edit. Keeping const on the filter is good practice, since it makes sure the object isn't accidentally overwritten, as it would raise a compiler error and get detected before it becomes a bug.